src/bot/commands/campaign.ts

```diff
@@ -101,16 +163,32 @@ const command: ICommand = {
             const campaignRepository = getRepository(Campaign);
             const result = await campaignRepository.find({
               where: { roomId: room.id },
+              relations: ['characters'],
             });
+
             if (result.length > 0) {
-              if (result[0].characters) {
-                result[0].characters.push(matchedChar);
-              } else {
-                const characters = [matchedChar];
-                result[0].characters = characters;
+              const campaign = result[0];
+              const alreadyJoined = campaign.characters.filter((e) => e.name === matchedChar.name).length > 0;
+
+              if (!alreadyJoined) {
+                const currentMap = JSON.parse(campaign.dungeon);
+                const myPos = JSON.parse(matchedChar.position);
+
+                myPos.x = currentMap.enter.x;
+                myPos.y = currentMap.enter.y;
+
+                matchedChar.position = JSON.stringify(myPos);
```

Comment:
I saw this line appearing elsewhere, should we consider passing these kinds of transformations in a utility or in a more smart way? So no one can forget to sanitize the input before saving. Maybe even just a function that prepares data for saving per type of entity